### ISSUE TITLE
Inconsistent behaviour adding the same job twice or updating without any change.

### ISSUE THREAD

**Issue by blmhemu** [hashicorp/nomad#10981]
### Nomad version
1.1.3

### Operating system and Environment details
Ubuntu 21.04 arm64

### Issue
I am trying to run the same job (Task1 see below) twice on nomad. Nomad still shows it is being updated (see pic1). This happens only for some tasks and seems inconsistent (see Pic1 and Pic2).

Task1:
```
job "redis" {
  datacenters = ["dc1"]
  type = "service"
  group "redis" {
    count = 1
    task "redis" {
      driver = "docker"
      config {
        image = "redis:6-alpine"
      }
      resources {
        cpu    = 200
        memory = 128
      }
    }
  }
}
```
Pic1:
![image](https://user-images.githubusercontent.com/19410501/127770187-4811fddb-aa8a-4d57-bad5-500b003c355e.png)

Task2:
```
A big traefik job
```

Pic2:
![image](https://user-images.githubusercontent.com/19410501/127771426-7a2ee5bb-9867-4db8-a080-966a1b2e6a73.png)

See the difference between pic1 and pic2.

Also when deploying using ansible (twice), I see some jobs changed while others remained the same. 
![image](https://user-images.githubusercontent.com/19410501/127770580-03bd2d8d-1e18-4ab1-8977-0bf47f86b0a7.png)

Previously (probably 0.10), this did not happen.

**Comment by blmhemu** [comment]
This can also be simulated by going to `Jobs->job (traefik)->Definition->Edit(Just click do not edit anything)->Save`

**Comment by blmhemu** [comment]
@notnoop Thanks for the quick fix! Will it be backported to 1.1.3 or will there be a new release (say 1.1.4) ?

**Comment by notnoop** [comment]
The change should be out in 1.1.4 (as well as 1.0.10), due to be released very soon. In my testing, this issue seems to only impact planning, but it doesn't seem to actually perform in-place updates. For example, when retrying your job in the CLI, I noticed that while `job plan` returned "forces in-place update" indicator, `job run` and subsequent `job status` didn't update the job version or make any updates:

```
$ ./nomad version
Nomad v1.1.3 (8c0c8140997329136971e66e4c2337dfcf932692)
$ ./nomad job run ./redis.nomad
==> 2021-08-02T12:07:11-04:00: Monitoring evaluation "ef5fa0dc"
    2021-08-02T12:07:11-04:00: Evaluation triggered by job "redis"
==> 2021-08-02T12:07:13-04:00: Monitoring evaluation "ef5fa0dc"
    2021-08-02T12:07:13-04:00: Evaluation within deployment: "e6fccfcf"
    2021-08-02T12:07:13-04:00: Allocation "533dca8d" created: node "27a22149", group "redis"
    2021-08-02T12:07:13-04:00: Evaluation status changed: "pending" -> "complete"
==> 2021-08-02T12:07:13-04:00: Evaluation "ef5fa0dc" finished with status "complete"
==> 2021-08-02T12:07:13-04:00: Monitoring deployment "e6fccfcf"
  ✓ Deployment "e6fccfcf" successful

    2021-08-02T12:07:24-04:00
    ID          = e6fccfcf
    Job ID      = redis
    Job Version = 0
    Status      = successful
    Description = Deployment completed successfully

    Deployed
    Task Group  Desired  Placed  Healthy  Unhealthy  Progress Deadline
    redis       1        1       1        0          2021-08-02T12:17:23-04:00
$ ./nomad job status redis
ID            = redis
Name          = redis
Submit Date   = 2021-08-02T12:07:11-04:00
Type          = service
Priority      = 50
Datacenters   = dc1
Namespace     = default
Status        = running
Periodic      = false
Parameterized = false

Summary
Task Group  Queued  Starting  Running  Failed  Complete  Lost
redis       0       0         1        0       0         0

Latest Deployment
ID          = e6fccfcf
Status      = successful
Description = Deployment completed successfully

Deployed
Task Group  Desired  Placed  Healthy  Unhealthy  Progress Deadline
redis       1        1       1        0          2021-08-02T12:17:23-04:00

Allocations
ID        Node ID   Task Group  Version  Desired  Status   Created  Modified
533dca8d  27a22149  redis       0        run      running  26s ago  14s ago
$ ./nomad job plan redis.nomad
+/- Job: "redis"
+/- Task Group: "redis" (1 ignore)
  +/- Task: "redis" (forces in-place update)

Scheduler dry-run:
- All tasks successfully allocated.

Job Modify Index: 10
To submit the job with version verification run:

nomad job run -check-index 10 redis.nomad

When running the job with the check-index flag, the job will only be run if the
job modify index given matches the server-side version. If the index has
changed, another user has modified the job and the plan's results are
potentially invalid.
$ ./nomad job run ./redis.nomad
==> 2021-08-02T12:07:52-04:00: Monitoring evaluation "2602d546"
    2021-08-02T12:07:52-04:00: Evaluation triggered by job "redis"
    2021-08-02T12:07:52-04:00: Evaluation within deployment: "e6fccfcf"
    2021-08-02T12:07:52-04:00: Evaluation status changed: "pending" -> "complete"
==> 2021-08-02T12:07:52-04:00: Evaluation "2602d546" finished with status "complete"
==> 2021-08-02T12:07:52-04:00: Monitoring deployment "e6fccfcf"
  ✓ Deployment "e6fccfcf" successful

    2021-08-02T12:07:52-04:00
    ID          = e6fccfcf
    Job ID      = redis
    Job Version = 0
    Status      = successful
    Description = Deployment completed successfully

    Deployed
    Task Group  Desired  Placed  Healthy  Unhealthy  Progress Deadline
    redis       1        1       1        0          2021-08-02T12:17:23-04:00
$ ./nomad job status redis
ID            = redis
Name          = redis
Submit Date   = 2021-08-02T12:07:11-04:00
Type          = service
Priority      = 50
Datacenters   = dc1
Namespace     = default
Status        = running
Periodic      = false
Parameterized = false

Summary
Task Group  Queued  Starting  Running  Failed  Complete  Lost
redis       0       0         1        0       0         0

Latest Deployment
ID          = e6fccfcf
Status      = successful
Description = Deployment completed successfully

Deployed
Task Group  Desired  Placed  Healthy  Unhealthy  Progress Deadline
redis       1        1       1        0          2021-08-02T12:17:23-04:00

Allocations
ID        Node ID   Task Group  Version  Desired  Status   Created  Modified
533dca8d  27a22149  redis       0        run      running  47s ago  35s ago
```
